### ISSUE TITLE
Add rapeseed to farming

### ISSUE THREAD

**Issue by Sokomine** [minetest/minetest_game#1679]
Dyes got too rare with recent changes in MTG. We also have nothing that provides oil.

Oilseed rape/rapeseed fields in full bloom are beautiful in RL. I'm sure a texture can be created that adds a plant that forms such nice fields.
Food mods usually require some oily/fat substance as that's usually part of the human diet in one way or another. Worlds with mobs can get butter from their cows. If there are no cows, there are sometimes crazy receipes like 2 sand -> one butter (probably because the sand has the right color - and there really isn't anything that fits any better).
Oil is also a base for oil colors. Admittedly those are diffrent from our dyes which occour in form of pulver. But then, getting dye as a pulver from a flower will require several steps instead of one crafting in RL as well, so there's no need to stay too realistic. Flower -> dye pulver and dye pulver + oil bottle -> 2 dye pulver would help builders a lot and offer a new way and incentive to do farming.

General idea:
Get the first oilseed seeds from dry shrub (it's there and not used for a plant yet) the same way junglegrass gives cotton seeds and grass gives wheat seeds.
The seeds then grow into oilseed rape with normal farming.
Harvesting fully grown oilseed rape gives more seeds - the seeds are the product. They can be used for planting more rape - or be cooked into oil (VanessaE has a nice texture of an oil bottle).
Dye pulver + oil would yield 2 dye pulver.

**Comment by Sokomine** [comment]
Forgot: Oil is also pretty useful for anything machine-related - even if it's only part of the craft receipe. Also useful as a fuel.

**Comment by VanessaE** [comment]
Suggest using the name "canola oil" or just "oil".  "rapeseed" may be the correct name, but it would be a poor choice of wording for a game. :)

**Comment by VanessaE** [comment]
Also consider that homedecor and pipeworks get oil from leaves.

**Comment by C1ffisme** [comment]
What about [hemp](https://en.wikipedia.org/wiki/Hemp)? Hemp fiber is used in making rope, and can apparently be made into [oil](https://en.wikipedia.org/wiki/Hemp_oil) which can seemingly be used as an alcohol-based fuel as well as paints and inks. Hemp seeds can also be eaten.

It might be a little close to cotton, though.

**Comment by VanessaE** [comment]
Hemp is good too.

**Comment by Sokomine** [comment]
Thanks for providing a better name, VanessaE. So let's call the oil just oil. A general term is better for usage in other mods anyway. Can the plant be called "canola"?
@C1ffisme: Hemp is fine for getting materials for clothes and the like, but we do have cotton for that already, and hemp fields seem to be mostly green. Rapeseed fields are a nice yellow. Also rapeseed/canola is grown specificly for oil production.

**Comment by VanessaE** [comment]
Wikipedia suggests that "canola" is a legit name for the plant itself, but only one specific variety.  Better to just call it "oil" and get it from leaves the way it's already done in other mods.

**Comment by sofar** [comment]
Thoughts:

- oil + dye could make paints. Right now UD allows smearing dye to blocks, but it really should be using paints, not dye powder.

- `coleseed` is an alternative name for `rapeseed`. The yellow color is would be a nice addition. We could just naturally grow it in certain biomes.

- I'd want to see a complementary set of uses in-game for oil. So allowing a bucket of oil to be fuel in a furnace would be good. Putting oil in flasks would be good too, so that perhaps it could be used for some type of food, or oil lamp. Of course, those conflict with the newly added meselamps and we already have enough lamps as is.

**Comment by VanessaE** [comment]
At one time, they were liquids in bottles, actually, with white paint being in a bucket.  You'd cook flowers, stone, or coal to get the colors.

It was certainly more realistic, but the idea lost traction when the default dyes mod was created.

**Comment by Sokomine** [comment]
Maybe a white bottle could be added. The colored bottles in unified_dyes where pretty nice. They could replace the images of the powders, or we can keep the current powders and pretend there's a step inbetween where the liquid dye gets dried. It's not too far-fetched as the step from flowers to powder would likewise include such a step. Crafting is a very powerful tool.

coleseed is too generic. Wikipedia describes cole as "A brassica, especially cabbage, kale, or rape (rapeseed)." While these three may belong to the same plant family, they're used very very diffrently as food. It is not at all obvious to regular consumers of cabbage and kale and observers of rapeseed fields that rape would belong to the same group of plants. Rape seems to be a plant that is too specific to Europe and only occours as canola in some english speaking countries. As MT is international, I'd suggest to call it just "oil plant" and "oil plant seeds". Players from around the world will then be able to understand its purpose and be able to think of it as whatever their native oil production plant would be. If all fails, rapeseed fields and sunflower fields (which can also be used for oil production) may look relatively similar from sufficiently far away - both are mostly yellow with green below. Sunflowers have more brownish color in the center, but then, at 16x16 pixel, there is more room for imagination than certainity. Which is good for the game.

Buckets of oil would be odd. Glass bottles will look better. The main purpose will be to make dyes available - which is important for people who want to use the game to build structures. Fuel is another useful purpose. Making them eatible is also reasonable. Further uses inside MTG are hard to think of because MTG is rather limited. Machines, even armor (polish it to make it look better),  food, creating plastic, soap, candles - there's a whole bunch of uses for oil which would be tricky to absurd to craft from other, non-oily sources. Pour it on the ground so that mobs and players may slip on it, pour it somewhere and ignite, throw oil bottles at mobs...but all that requires further content from other mods. Still, the "oil poured out" (ignitable with flint) might work on its own (player physics?) and look good. Blocks of fat would also be possible but maybe not look well enough to be worth the trouble.

**Comment by gpcf** [comment]
Or add an oil press to press stuff. I think the best idea would just be to add an oil group and then have mods add stuff. I'm envisioning mods adding canola and olive trees and oil presses, including only cotton in the default.

**Comment by C1ffisme** [comment]
> As MT is international, I'd suggest to call it just "oil plant" and "oil plant seeds"

So, for the same reason, we should rename cotton to "fibrous plant" and wheat to "grain"? Why? It might be considered educational if we're encouraging people to look up a plant on Wikipedia. We might as well rename all the trees, then, because some people don't know what they are. :P

**Comment by Fixer-007** [comment]
This PR can probably be used when automation will be added (if ever)

**Comment by C1ffisme** [comment]
@Fixer-007 This isn't a PR...

**Comment by paramat** [comment]
Any core dev support?
Personally i think the idea is fine for an optional mod but i'm unsure if it's too specialist for MTG. In these situations i prefer to choose simplicity, as we are shifting focus to new games. So i'm leaning towards :-1:

**Comment by SmallJoker** [comment]
Cotton and wheat alone get quite boring after a while, so I'd welcome more plant types - preferably such that are not contained in another mod yet.

**Comment by paramat** [comment]
Thanks for input, to be clear i have no objection if other core devs support.

**Comment by sofar** [comment]
I'm entirely fine with more farming content (hence: `crops`). I do believe that there should be a well written inclusion that makes sense from a crafting /gameplay perspective. Essentially that is the dilemma, I think. I also think that the name is inappropriate in this case, but that's not unfixable.

**Comment by paramat** [comment]
There is stated support for more crops, but, no-one has specifically supported adding rapeseed. Do you?

**Comment by Ezhh** [comment]
I'm fine with it in principle, but agree with Vanessa's comment that another name would be better. 

What can be done with it would need to be carefully thought about. I don't think we should add something like this unless we also add a use for it.

**Comment by paramat** [comment]
Yes i think a use is necessary, and a needed use, not just finding a use to justify adding.
Currently i don't see a needed use. A mod can add rapeseed if oil is needed.
So i am somewhat -1 to avoid low priority complexity.

**Comment by SmallJoker** [comment]
@paramat I am currently downvoting your recent comment because MTG clearly lacks some variety in farming. Rapeseed could be a good addition, unlike potatoes or tomatoes which are contained in many other farming mods.

**Comment by Ezhh** [comment]
Agreed with @SmallJoker subject to my comments above. I support adding this, just using a different name and making sure there is a use for it.

**Comment by DS-Minetest** [comment]
>Yes i think a use is necessary

Imo things have a right of existence even if they don't have a use. There's weed for example. Also you don't need flies. But we humans normally don't exterminate everything that doesn't have any use for us. What are ice bears good for? And what about handicapped persons who are not able to do the same work as healthy ones? Do we kill them? Of course not. Every form of life is valuable. So is every thing that stores information. Exaggeration ends here.
Rapeseed could look good and colorful as plant itself and so be used as decoration. It would have nearly as much use as cactus, stone brick or bookshelf. Also with more things, that the player can farm, his playing experience will be much greater and he will have more fun as he is able to do more.

**Comment by Ezhh** [comment]
@DS-Minetest That comment is just wrong. Exaggerated or otherwise.

To add some clarification - I'm not objecting to things having a decorative use, but this is something specifically suggested as a farmable crop. Farmable crops need a use or you don't farm them.

**Comment by TumeniNodes** [comment]
Two years later... (almost)
You know... this new oil _could_ be used to soak into wooden tools to repair them a bit, add some strength to them? (as in real life) soaking wooden handles with oil keeps them from drying out and becoming brittle, which strengthens them and adds longevity.

So that's one use.
I'm sure I or others can think of a couple more...
But maybe just this one for now might be enough to move forward with this idea? after idling for so long

**Comment by paramat** [comment]
Adding rapeseed crops and a food oil resource seems ok to me.

However i disagree with the proposed usage for multiplying dye. The limited dye problem should be solved in a simpler, cleaner and less awkward way, either by flowers dropping more dye or making flowers farmable using seeds.
Oilseed rape needs to be justified by other uses, and it probably is.

I also disagree with getting the first seeds from dry shrub, this is a messy 'old MTG' way to get initial seeds, we should have a wild oilseed rape plant, just like cotton will soon be.

**Comment by Desour** [comment]
>I also disagree with getting the first seeds from dry shrub, this is a messy 'old MTG' way to get initial seeds, we should have a wild oilseed rape plant, just like cotton will soon be.

- Imo getting seeds from grass is one of the few MTG mechanics that already exist for a long time but are not problematic. (Eg. I haven't yet heard any complaints about getting seeds from grass, but there are many complaints about the uselessness of bronze and the difficulty to get flowers (or the fact that dye is essential for some things like beds).)
- Getting seeds from grass gives grass a use, which is great. Otherwise all grass would be useless, like dry shrub currently is.
- Realism doesn't matter.
- If you later add new plants for getting seeds, players in old maps won't be able to get any seeds and hence don't have the ability to do farming (unless they already have seeds, which won't be the case for rapeseed).
This will furthermore prevent us from adding real uses to rapeseed (i.e. uses that make it obligatory to do farming before getting the best available tools).

TL;DR: Getting initial rapeseed from dry shrub is much better than adding some new wild rapeseed plant.

**Comment by Desour** [comment]
Some ideas for usecases:
- >oil could be used to soak into wooden tools ...

  Imo it would be a nice feature if you needed something like oily sticks for any tools that are better than bronze (steel, mese, diamond). This would even make bronze useful. And this would force the player to farm, which increases the little time that you currently need to get from bare hands to mese tools.
- >added into the torches recipe

  What about an alternative, sustainable crafting recipe for torches?
- >i disagree with the proposed usage for multiplying dye. [...] flowers dropping more dye

  I suggest adding dye recipes that take the corresponding dye and oil to get more dye than in the recipe where one just uses flowers. This would mean no perpetual dye but oil could still be used for dye production.

**Comment by paramat** [comment]
> Eg. I haven't yet heard any complaints about getting seeds from grass,

I have been long complaining about getting seeds from what are mostly unrelated species instead of a dedicated wild plants. My wild cotton PR is supported.
Lack of complaints does not imply lack of a problem, players tend to get used to how MTG is.
> Getting seeds from grass gives grass a use, which is great. Otherwise all grass would be useless, like dry shrub currently is.

Grass and dry shrub are not 'useless'. This annoying repeated argument is silly as it would lead to removing very many nodes that make the world far more attractive and enjoyable.
Just because getting cotton seed from junglegrass is a 'use' does not make it acceptable, because it is silly.
> Realism doesn't matter.

This is clearly nonsense, MTG is based on realism and is loosely realistic, it is abstracted and simplifed realism.
> If you later add new plants for getting seeds, players in old maps won't be able to get any seeds and hence don't have the ability to do farming

They only need to venture a little way into new terrain to find 1 seed item.
When new nodes or items are added to MTG they often are only in new terrain, this is normal and acceptable.
> I suggest adding dye recipes that take the corresponding dye and oil to get more dye than in the recipe where one just uses flowers.

That is essentially the original suggestion that i disagree with.

**Comment by Desour** [comment]
> Grass and dry shrub are not 'useless', they make the world far more attractive and enjoyable.

Well, the world might look even more beautiful, but it will consist of even less usable resources (=resources that players can process to something that makes them advance in the game, as opposed to items that just fill the inventory).

> > Realism doesn't matter.
> 
> This is clearly nonsense, MTG is based on realism and is loosely realistic, it is abstracted and simplifed realism.

What I wanted to say here is that it is no problem that grass is not of the same species as the plant that grows out of the seeds that get dropped eventually (or at least a related species).
More realism in this case doesn't really lead to more fun at playing the game (ie. the player doesn't care much if they have to dig the one plant or the other).

> > I suggest adding dye recipes that take the corresponding dye and oil to get more dye than in the recipe where one just uses flowers.
> 
> That is essentially the original suggestion that i disagree with.

Oops, I've meant to write flower, not dye. Corrected in original post.

**Comment by Desour** [comment]
PR: #2626
